### PR TITLE
Custom resize cursors

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Moves the sort icon to the start of its header cell
Shows a chevron indicating a menu on column headers for hover/focus, this can be seen on any resizable column or any resizable and sortable column
Displays a custom cursor for resizing

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by devongovett** [comment]
What's the plan for the aria hooks? Will we somehow fall back to the standard ones? Embed these?

File: packages/@adobe/spectrum-css-temp/components/table/index.css [hunk 1]
```diff
@@ -455,11 +455,20 @@ svg.spectrum-Table-sortedIcon {
...
+}
+.spectrum-Table-columnResizer--ewresize {
+  cursor: url('./cursors/Cur_MoveHorizontal_9_9.svg'), ew-resize;
+}
+.spectrum-Table-columnResizer--eresize {
+  cursor: url('./cursors/Cur_MoveToRight_9_9.svg'), e-resize;
+}
+.spectrum-Table-columnResizer--wresize {
+  cursor: url('./cursors/Cur_MoveToLeft_9_9.svg'), w-resize;
 }
```
Comment (devongovett) [inline packages/@adobe/spectrum-css-temp/components/table/index.css:474]: We'll somehow need to embed these as base64. I'm not sure we can rely on people's bundlers being able to handle these assets, and we don't want a network delay when showing them.

**Comment by snowystinger** [comment]
> What's the plan for the aria hooks? Will we somehow fall back to the standard ones? Embed these?

I decided to move it out entirely, there's enough information coming from the hook to set whatever cursor you want. Reason being that it seemed like a rendering concern more than aria. Especially since you need to set the cursor on hover as well as setting the entire document, every element, with !important during drag. Without creating a stylesheet on the fly, setting it for every element would be impossible.

File: packages/@adobe/spectrum-css-temp/components/table/index.css [hunk 2]
```diff
@@ -106,6 +115,12 @@ svg.spectrum-Table-sortedIcon {
       text-overflow: ellipsis;
     }
   }
+  &.is-hovered,
+  &.is-cell-focused {
+    .spectrum-Table-menuChevron {
+      display: inline-block;
```
Comment (devongovett) [inline packages/@adobe/spectrum-css-temp/components/table/index.css:121]: Is there supposed to be a menu chevron on hover/focus? I don't see one.

Comment (snowystinger) [reply packages/@adobe/spectrum-css-temp/components/table/index.css:121]: O, I see, it's only showing for columns that have both resize AND sort, I'll get that sorted

**Comment by ktabors** [comment]
The most recent storybook v18 build isn't showing the new cursor.

**Review by ktabors — APPROVED** [review]
LGTM

**Review by LFDanLu — COMMENTED** [review]
The behavior seems good to me, but I'm not sure if https://github.com/adobe/react-spectrum/pull/3535#discussion_r974787256 is blocking/resolved (just not even sure what exactly it means heh).

Some extra things that are out of scope/can be done as followup/maybe non-issues:
1. Can/should we bump up the default min width of a resizable column? Now that we have a  chevron some of the story columns are quite cramped at the min width. 
image

This maybe a non-issue since the user could define a nicer min width themselves but perhaps we can approximate a new default min width value that is a 3 or 4 letter word + chevron

2. Should these column header buttons be truncating text? They currently don't. It would also be nice if it was like Picker where the chevron was prioritized over the text, at least when it is visible (display: none when not hovered?)

**Comment by ktabors** [comment]
I didn't consider what Daniel reported as an issue, but since we are this doesn't seem right. Give it keyboard focus and then mouse over it.
image

**Comment by snowystinger** [comment]
> I didn't consider what Daniel reported as an issue, but since we are this doesn't seem right. Give it keyboard focus and then mouse over it. image

Will be fixed by https://github.com/adobe/react-spectrum/pull/3560

**Review by ktabors — APPROVED** [review]
This works well and looks good!

**Review by reidbarber — APPROVED** [review]
LGTM

**Pull request merged**